teerex/src/lib.rs

```diff
@@ -162,70 +169,147 @@ pub mod pallet {
 		// the integritee-service wants to register his enclave
 		#[pallet::call_index(0)]
 		#[pallet::weight((<T as Config>::WeightInfo::register_ias_enclave(), DispatchClass::Normal, Pays::Yes))]
-		pub fn register_ias_enclave(
+		pub fn register_sgx_enclave(
 			origin: OriginFor<T>,
-			ra_report: Vec<u8>,
-			worker_url: Vec<u8>,
+			proof: Vec<u8>,
+			worker_url: Option<Vec<u8>>,
+			attestation_method: SgxAttestationMethod,
 		) -> DispatchResultWithPostInfo {
-			log::info!("teerex: called into runtime call register_ias_enclave()");
+			log::info!("teerex: called into runtime call register_sgx_enclave()");
 			let sender = ensure_signed(origin)?;
-			ensure!(ra_report.len() <= MAX_RA_REPORT_LEN, <Error<T>>::RaReportTooLong);
-			ensure!(worker_url.len() <= MAX_URL_LEN, <Error<T>>::EnclaveUrlTooLong);
+			ensure!(proof.len() <= SGX_RA_PROOF_MAX_LEN, <Error<T>>::RaProofTooLong);
+			ensure!(
+				worker_url.clone().unwrap_or_default().len() <= MAX_URL_LEN,
+				<Error<T>>::EnclaveUrlTooLong
+			);
```

Comment:
Unnecessary clone can be removed.

Maybe `worker_url.as_ref()` needed in the `if let` statement.
```suggestion
if let Some(url) = worker_url {
	ensure!(url.len() <= MAX_URL_LEN, <Error<T>>::EnclaveUrlTooLong);
}
```